### PR TITLE
zpool: Change base URL for ZFS messages to openzfs-docs

### PR DESCRIPTION
**Pull request by freqlabs**
### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [x] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [ ] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

File: module/zfs/spa.c [hunk 1]
```diff
@@ -3201,7 +3201,8 @@ spa_verify_host(spa_t *spa, nvlist_t *mos_config)
 			cmn_err(CE_WARN, "pool '%s' could not be "
 			    "loaded as it was last accessed by "
 			    "another system (host: %s hostid: 0x%llx). "
-			    "See: http://illumos.org/msg/ZFS-8000-EY",
+			    "See: http://openzfs.github.io/openzfs-docs/msg/"
```
Comment (behlendorf) [inline module/zfs/spa.c:3204]: http -> https

**Comment by gmelikov** [comment]
I've found several more `zfsonlinux.org/msg` in code, let's change them too.

**Review by Ornias1993 — APPROVED** [review]
Good catch!

**Comment by ceedriic** [comment]
In the root of the openzfs/zfs tree, I guess at least NEWS and probably AUTHORS needs an update...

**Comment by freqlabs** [comment]
> In the root of the openzfs/zfs tree, I guess at least NEWS and probably AUTHORS needs an update...

I'm just going for the msg/ZFS-* URLs in this PR, but the NEWS link is a good catch. I don't see anything in AUTHORS that should be changed.

Can you open a PR to update the URL in NEW?

**Review by gmelikov — APPROVED** [review]

**Comment by ceedriic** [comment]
> > In the root of the openzfs/zfs tree, I guess at least NEWS and probably AUTHORS needs an update...
> 
> I'm just going for the msg/ZFS-* URLs in this PR, but the NEWS link is a good catch. I don't see anything in AUTHORS that should be changed.

I cannot find mattmacy in it for example.
 
> Can you open a PR to update the URL in NEWS?

#10824

**Pull request merged**